gateway/ln-gateway/src/lnrpc_client.rs

```diff
@@ -72,64 +65,109 @@ impl NetworkLnRpcClient {
             GatewayError::Other(anyhow!("Failed to create lnrpc endpoint from url"))
         })?;
 
-        let client = GatewayLightningClient::connect(endpoint)
-            .await
-            .map_err(|e| {
-                error!("Failed to connect to lnrpc server: {:?}", e);
-                GatewayError::Other(anyhow!("Failed to connect to lnrpc server"))
-            })?;
-
-        Ok(Self { client })
+        let mut gw_rpc = NetworkLnRpcClient {
+            client: None,
+            endpoint,
+        };
+        gw_rpc.connect().await?;
+        Ok(gw_rpc)
     }
 }
 
 #[async_trait]
 impl ILnRpcClient for NetworkLnRpcClient {
     async fn pubkey(&self) -> Result<GetPubKeyResponse> {
-        let req = Request::new(EmptyRequest {});
+        if let Some(mut client) = self.client.clone() {
+            let req = Request::new(EmptyRequest {});
+            let res = client.get_pub_key(req).await?;
 
-        let mut client = self.client.clone();
-        let res = client.get_pub_key(req).await?;
+            return Ok(res.into_inner());
+        }
 
-        Ok(res.into_inner())
+        error!("Error: not connected to CLN extension.");
+        Err(GatewayError::Other(anyhow!(
+            "Error not connected to CLN extension"
+        )))
     }
 
     async fn routehints(&self) -> Result<GetRouteHintsResponse> {
-        let req = Request::new(EmptyRequest {});
+        if let Some(mut client) = self.client.clone() {
+            let req = Request::new(EmptyRequest {});
+            let res = client.get_route_hints(req).await?;
 
-        let mut client = self.client.clone();
-        let res = client.get_route_hints(req).await?;
+            return Ok(res.into_inner());
+        }
 
-        Ok(res.into_inner())
+        error!("Error: not connected to CLN extension.");
+        Err(GatewayError::Other(anyhow!(
```

Comment:
Maybe make a `fn GatewayError:other(msg: &str) -> Self` since it repeats many times.

Reply:
Done